crypto/evp/kem.c

```diff
@@ -87,6 +87,7 @@ static int evp_kem_init(EVP_PKEY_CTX *ctx, int operation,
          * iteration we're on.
          */
         EVP_KEM_free(kem);
+        kem = NULL;
```

Comment:
`kem` is assigned a value in the switch below. It is not used in between here and the assignment. Please note that `iter` can really have value of 1 or 2 only. For this reason the assignment is useless.